cmd/gateway/main.go

```diff
@@ -100,24 +100,58 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 	if !accessKeyFlag.Changed {
 		accessKey, err := generateKey()
 		if err != nil {
-			return err
+			return Error.Wrap(err)
 		}
+
 		overrides[accessKeyFlag.Name] = accessKey
 	}
+
 	secretKeyFlag := cmd.Flag("minio.secret-key")
 	if !secretKeyFlag.Changed {
 		secretKey, err := generateKey()
 		if err != nil {
-			return err
+			return Error.Wrap(err)
 		}
+
 		overrides[secretKeyFlag.Name] = secretKey
 	}
 
-	if !setupCfg.NonInteractive {
-		return setupCfg.interactive(cmd, setupDir, overrides)
+	// override is required because the default value of Enc.KeyFilepath is ""
+	// and setting the value directly in setupCfg.Enc.KeyFiletpathon will set the
+	// value in the config file but commented out.
+	encryptionKeyFilepath := setupCfg.Enc.KeyFilepath
+	if encryptionKeyFilepath == "" {
+		encryptionKeyFilepath = filepath.Join(setupDir, ".encryption.key")
+		overrides["enc.key-filepath"] = encryptionKeyFilepath
 	}
 
-	return process.SaveConfigWithAllDefaults(cmd.Flags(), filepath.Join(setupDir, "config.yaml"), overrides)
+	if setupCfg.NonInteractive {
+		return setupCfg.nonInteractive(cmd, setupDir, encryptionKeyFilepath, overrides)
+	}
+
+	return setupCfg.interactive(cmd, setupDir, encryptionKeyFilepath, overrides)
+}
+
+func (flags GatewayFlags) nonInteractive(
+	cmd *cobra.Command, setupDir string, encryptionKeyFilepath string, overrides map[string]interface{},
+) error {
+	if setupCfg.Enc.EncryptionKey != "" {
+		err := uplink.SaveEncryptionKey(setupCfg.Enc.EncryptionKey, encryptionKeyFilepath)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+	}
+
+	err := process.SaveConfigWithAllDefaults(cmd.Flags(), filepath.Join(setupDir, "config.yaml"), overrides)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	if setupCfg.Enc.EncryptionKey != "" {
+		_, _ = fmt.Printf("Your encryption key is saved to: %s\n", encryptionKeyFilepath)
```

Comment:
if we aren't using the return value, why do we need this assignment here?

Reply:
I mostly assign return values to the blank identifier when I want to highlight that I'm aware of the returned but I want them to be ignored; that's very important for sensitive return value like `error` where they shouldn't almost be unhandled.

However I don't usually apply it for calls `fmt.Print..` functions, but I did here because in the implementation that I found (and changed in this branch) it used to call the `fmt.Print...` functions assigning the return values to the blank identifier, and I don't see a reason why not to make my changes respecting such style.